scripts/taxonomy/metadata_models.py

```diff
@@ -0,0 +1,318 @@
+import collections
+import logging
+import json
+import os
+from isb_web import config
+
+from isamples_metadata.Transformer import Transformer
+from scripts.taxonomy.Model import Model
+from scripts.taxonomy.SESARClassifierInput import SESARClassifierInput
+from scripts.taxonomy.OpenContextClassifierInput import OpenContextClassifierInput
+
+
+class MetadataModelLoader:
+    """Class that instantiates the pretrained models"""
+
+    # loaded models that will be used for classification
+    _SESAR_MATERIAL_MODEL = None
+    _OPENCONTEXT_MATERIAL_MODEL = None
+    _OPENCONTEXT_SAMPLE_MODEL = None
+
+    @staticmethod
+    def load_model_from_path(collection, label_type):
+        """
+            Set the pretrained models by loading them from the file system
+            Prerequisite: In order to use this, make sure that there is a pydantic settings file on the
+            at the root of this repository named "isamples_web_config.env" with the config file path, and the model path
+            defined
+
+            :param collection : the collection type of the sample
+            :param label_type : the field that we want to predict
+        """
+        # load the model config file
+        if collection == "SESAR" and label_type == "material":
+            config_path = config.Settings().sesar_material_config_path
+        elif collection == "OPENCONTEXT" and label_type == "material":
+            config_path = config.Settings().opencontext_material_config_path
+        elif collection == "OPENCONTEXT" and label_type == "sample":
+            config_path = config.Settings().opencontext_sample_config_path
+        # read the model config file
+        if not os.path.exists(config_path):
+            logging.error(
+                "Unable to locate pretrained models",
+            )
+            return
+        with open(config_path) as json_file:
+            config_json = json.load(json_file)
+        # use the model config to get the pretrained model
+        model = Model(config_json)
+
+        # initialize the model fields
+        if collection == "SESAR" and label_type == "material":
+            MetadataModelLoader._SESAR_MATERIAL_MODEL = model
+        elif collection == "OPENCONTEXT" and label_type == "material":
+            MetadataModelLoader._OPENCONTEXT_MATERIAL_MODEL = model
+        elif collection == "OPENCONTEXT" and label_type == "sample":
+            MetadataModelLoader._OPENCONTEXT_SAMPLE_MODEL = model
+
+    @staticmethod
+    def initialize_models():
```

Comment:
I think the idea of this is good, but there's nothing that requires this method to be called.  The best solution might be to make each of the individual getter methods e.g. `get_xxx_model()` to do a lazy check to make sure it isn't `None` and then call `initialize_models` if it *is* `None`.  At a minimum we want some code that says you need to call `initialize_models()` before attempting to use this, or we're silently fail.